arch/arm/core/aarch32/Kconfig

```diff
@@ -14,7 +14,7 @@ config CPU_CORTEX_M
 	select ARCH_HAS_TRUSTED_EXECUTION if ARM_TRUSTZONE_M
 	select ARCH_HAS_STACK_PROTECTION if (ARM_MPU && !ARMV6_M_ARMV8_M_BASELINE) || CPU_CORTEX_M_HAS_SPLIM
 	select ARCH_HAS_USERSPACE if ARM_MPU
-	select ARCH_HAS_NOCACHE_MEMORY_SUPPORT if ARM_MPU && CPU_HAS_ARM_MPU && CPU_CORTEX_M_HAS_CACHE
+	select ARCH_HAS_NOCACHE_MEMORY_SUPPORT if ARM_MPU && CPU_HAS_ARM_MPU && DCACHE
```

Comment:
I have mixed feeling about that change. This makes `nocache` memory available only if the cache is enabled. While it doesn't make a lot of sense to have `nocache` memory with the data cache disabled, it might prevent to simply set `DCACHE=n` for testing.

Reply:
good point. I would probably take that out entirely.

Reply:
on a second thought I make that depending on `CPU_HAS_DCACHE`

Reply:
Sounds good, thanks.